File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/LineSource/freeLine_CervenyB.py
#==================================================================
#  
#  Bellhop: Free space, line source, Cerveny Gaussian beam
#  Mexilhoeira Grande, Dom Jul 30 16:28:16 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run freeLine_CervenyB

from os import system
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readshd import *

rs = 0.0

print("Bellhop - Free space, line source, Cerveny Gaussian beam:")
print("Coherent TL calculation")
print("Ray centered beams")

system("bellhop.exe freeLine_CervenyB")

filename = 'freeLine_CervenyB.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure, axis=(0,1) )
tl = -20*log10( abs( p ) )

figure(1)
imshow(tl,extent=[0,rmaxkm,0,Dmax],aspect='auto',cmap='jet_r',origin='lower')
cb = colorbar()
cb.ax.invert_yaxis()
plot(rs,zs,marker="<",markersize=16,color="k")
xlabel('Range (km)')
ylabel('Depth (m)')
title('Bellhop - Free space, line source, Cerveny Gaussian beam')
xlim(0,rmaxkm)
ylim(Dmax,0)

show()

print("done.")

File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/LineSource/freeLine_CervenyB.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/Plot
addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('Bellhop - Free space, line source') 
disp('Coherent TL calculation')
disp('Ray centered beams')

system('bellhop.exe freeLine_CervenyB');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'freeLine_CervenyB.shd' );

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

tej = flipud( jet( 256 ) );

figure(1), hold on
plot(0,zs,'ko',0,zs,'m*','MarkerSize',16) 
pcolor(rarraykm,zarray,tl), shading interp, colormap( tej ), %caxis([0 20]), 
h = colorbar;
set( h, 'YDir', 'reverse' );
axis([0 rmaxkm 0 zmax])
hold off
view(0,-90)
xlabel('Range (km)')
ylabel('Depth (m)')
title('Bellhop - Free space, line source, 5 Hz')

disp('done.')
